harness/harness-core-ui
# fix: [CDS-83638]: package dropdown for GAR

### Summary
Enhancement - Adding support of package field in GAR (https://harness.atlassian.net/browse/CDS-83638)
Solution - Added support of package in GAR in pipeline/runform/trigger.



Manual testing - 

Create form 

- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v1 api is called

Runtime form

- Package field is getting rendered when its made runtime in pipeline.
- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v2 api is called

Template form
- Changes made in Package field are reflected in yaml and unsaved changes doesnt appear once saved.
- Package is getting fetched when all required fields are fixed and have value.
- Package api call is not made when any of required field is runtime/expression or has empty fixed value.
- Clearing Package field when any of the required field is changed.
- Clearing Package dropdown list when any of the required field is changed.
- Package v1 api is called






#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

https://github.com/harness/harness-core-ui/assets/96036463/22d74489-745a-4c42-a627-48cb911b6c29



## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)
